main/src/org/jetbrains/bsp/bazel/install/Install.java

```diff
@@ -0,0 +1,214 @@
+                Constants.NAME,
+                argv,
+                Constants.VERSION,
+                Constants.BSP_VERSION,
+                Constants.SUPPORTED_LANGUAGES);
+
+        Path rootDir =
+            cmd.hasOption(DIRECTORY_SHORT_OPT)
+                ? Paths.get(cmd.getOptionValue(DIRECTORY_SHORT_OPT))
+                : Paths.get("");
+
+        writeConfigurationFiles(rootDir, details);
+
+        System.out.println("Bazel BSP server installed in " + rootDir.toString());
+      }
+    } catch (ParseException e) {
+      writer.println(e.getMessage());
+      formatter.printUsage(writer, 120, INSTALLER_BINARY_NAME, cliOptions);
+      hasError = true;
+    } catch (NoSuchElementException e) {
+      writer.println(e.getMessage());
+      hasError = true;
+    } finally {
+      writer.close();
+    }
+
+    if (hasError) {
+      System.exit(1);
+    }
+  }
+
+  private static void writeConfigurationFiles(Path rootDir, Object discoveryDetails)
+      throws IOException {
+    Path bazelbspDir = rootDir.resolve(Constants.BAZELBSP_DIR_NAME);
+    Files.createDirectories(bazelbspDir);
+
+    // Copy aspects.bzl file
+    Path aspectsFile = bazelbspDir.resolve(Constants.ASPECTS_FILE_NAME);
+    Files.copy(
+        Install.class.getResourceAsStream(Constants.ASPECTS_FILE_NAME),
+        aspectsFile,
+        StandardCopyOption.REPLACE_EXISTING);
+
+    // Create an empty BUILD file, overwrite if exists
+    Files.newByteChannel(
+        bazelbspDir.resolve(Constants.BUILD_FILE_NAME),
+        StandardOpenOption.CREATE,
+        StandardOpenOption.WRITE);
+
+    Path bspDir = rootDir.resolve(Constants.BSP_DIR_NAME);
+    Files.createDirectories(bspDir);
+
+    // Write bazelbsp.json file which allows bsp server discovery
+    Files.write(
+        bspDir.resolve(Constants.BAZELBSP_JSON_FILE_NAME),
+        new GsonBuilder()
+            .setPrettyPrinting()
+            .create()
+            .toJson(discoveryDetails)
+            .getBytes(StandardCharsets.UTF_8));
+  }
+
+  private static Options getCliOptions() {
+    Options cliOptions = new Options();
+
+    Option java =
+        Option.builder(JAVA_SHORT_OPT)
+            .longOpt("java")
+            .hasArg()
+            .argName("path")
+            .desc("Use provided Java executable to run the BSP server")
+            .build();
+
+    cliOptions.addOption(java);
+
+    Option bazel =
+        Option.builder(BAZEL_SHORT_OPT)
+            .longOpt("bazel")
+            .hasArg()
+            .argName("path")
+            .desc("Make the BSP server use this Bazel executable")
+            .build();
+
+    cliOptions.addOption(bazel);
+
+    Option debug =
+        Option.builder(DEBUGGER_SHORT_OPT)
+            .longOpt("debugger")
+            .hasArg()
+            .argName("address (e.g. '127.0.0.1:8000'")
+            .desc("Allow BSP server debugging")
+            .build();
+
+    cliOptions.addOption(debug);
+
+    Option directory =
```

Comment:
It looks veeery similar to the `java` and `bazel` initialization. Can we extract it to the separate method?

Reply:
There's no need to, calling the extracted method would look almost the same - we still need to pass the arguments.